src/l1/mod.rs

```diff
@@ -43,7 +43,7 @@ pub struct ChainWatcher {
     /// The L2 starting block
     l2_start_block: u64,
     /// Channel for receiving block updates for each new block
-    pub block_update_receiver: mpsc::Receiver<BlockUpdate>,
+    block_update_receiver: Option<mpsc::Receiver<BlockUpdate>>,
```

Comment:
Nice change. This is a lot clearer than what we were doing before with leaving a useless channel there on initialization.